### PR TITLE
Removed Mailer component docs

### PR DESCRIPTION
**Pull request by wouterj**
See symfony/symfony#37542

The component is quite easy to use in a standalone project. The example in the component's README will be enough I think (no need to confuse framework users by adding the examples in this doc).

Meanwhile, I'm not sure what to do with the Mime component. Should we move it into a subguide (`/mailer/mime.rst`)? It doesn't really make sense to have it as a main guide I think.

### CODE REVIEW

File: mailer.rst [hunk 1]
```diff
@@ -785,6 +1016,41 @@ you have a transport called ``async``, you can route the message there:
             ],
         ]);
 
+    .. code-block:: php-standalone
```
Comment (wouterj) [inline mailer.rst:1019]: I'm not very sure about this code block as well. Seems very advanced to set-up message bus + mailer in a standalone application.

File: mailer.rst [hunk 2]
```diff
@@ -79,22 +79,85 @@ also have options that can be configured with query parameters at the end of the
...
+        MAILER_DSN=mailgun+https://KEY:DOMAIN@example.com
+        MAILER_DSN=mailgun+https://KEY:DOMAIN@example.com:99
+
+    Note that the protocol is *always* HTTPs and cannot be changed.
+
+High Availability
+~~~~~~~~~~~~~~~~~
+
+Symfony's mailer supports `high availability`_ via a technique called "failover"
+to ensure that emails are sent even if one mailer server fails .
```
Comment (xabbuh) [inline mailer.rst:125]: there's a space before the dot

File: mailer.rst [hunk 3]
```diff
@@ -79,22 +79,85 @@ also have options that can be configured with query parameters at the end of the
...
+
+The mailer will start using the first transport. If the sending fails, the
+mailer won't retry it with the other transports, but it will switch to the next
+transport automatically for the following deliveries.
+
+Load Balancing
+~~~~~~~~~~~~~~
+
+Symfony's mailer supports `load balancing`_ via a technique called "round-robin"
+to distribute the mailing workload across multiple transports .
```
Comment (xabbuh) [inline mailer.rst:142]: same here

File: mailer.rst [hunk 4]
```diff
@@ -503,15 +600,46 @@ called ``css`` that points to the directory where ``email.css`` lives:
...
+                http://symfony.com/schema/dic/twig https://symfony.com/schema/dic/twig/twig-1.0.xsd">
 
-        paths:
-            # point this wherever your css files live
-            '%kernel.project_dir%/assets/css': css
+            <twig:config>
+                <!-- ... -->
+
+                <!-- point this wherever your css files live -->
+                <twig:path namespace="images">%kernel.project_dir%/assets/css</twig:path>
```
Comment (xabbuh) [inline mailer.rst:629]: `namespace="css"`

File: mailer.rst [hunk 5]
```diff
@@ -503,15 +600,46 @@ called ``css`` that points to the directory where ``email.css`` lives:
...
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/twig.php
+        $container->loadFromExtension('twig', [
+            // ...
+            'paths' => [
+                // point this wherever your css files live
+                '%kernel.project_dir%/assets/css' => 'images',
```
Comment (xabbuh) [inline mailer.rst:640]: `'%kernel.project_dir%/assets/css' => 'css',`

File: mailer.rst [hunk 6]
```diff
@@ -816,14 +1008,51 @@ Always Send to the same Address
...
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <!-- ... -->
+            <framework:config>
+                <framework:mailer>
+                    <framework:envelope>
+                        <framework:envelope>youremail@example.com</framework:envelope>
```
Comment (xabbuh) [inline mailer.rst:1036]: `<framework:recipient>youremail@example.com</framework:recipient>`?

File: mailer.rst [hunk 7]
```diff
@@ -816,14 +1008,51 @@ Always Send to the same Address
...
+
+        # config/packages/dev/mailer.yaml
+        framework:
+            mailer:
+                envelope:
+                    recipients: ['youremail@example.com']
+
+    .. code-block:: xml
+
+        <!-- config/packages/framework.xml -->
```
Comment (xabbuh) [inline mailer.rst:1023]: `<!-- config/packages/dev/framework.xml -->`

File: mailer.rst [hunk 8]
```diff
@@ -816,14 +1008,51 @@ Always Send to the same Address
...
+                    <framework:envelope>
+                        <framework:envelope>youremail@example.com</framework:envelope>
+                    </framework:envelope>
+                </framework:mailer>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
```
Comment (xabbuh) [inline mailer.rst:1044]: `// config/packages/dev/framework.php`

File: mailer.rst [hunk 9]
```diff
@@ -798,12 +961,41 @@ While developing (or testing), you may want to disable delivery of messages enti
...
+    .. code-block:: yaml
+
+        # config/packages/dev/mailer.yaml
+        framework:
+            mailer:
+                dsn: 'null://null'
+
+    .. code-block:: xml
+
+        <!-- config/packages/framework.xml -->
```
Comment (xabbuh) [inline mailer.rst:975]: `<!-- config/packages/dev/mailer.xml -->`

File: mailer.rst [hunk 10]
```diff
@@ -798,12 +961,41 @@ While developing (or testing), you may want to disable delivery of messages enti
...
+
+            <!-- ... -->
+            <framework:config>
+                <framework:mailer dsn="null://null"/>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
```
Comment (xabbuh) [inline mailer.rst:992]: `// config/packages/dev/mailer.php`

File: mailer.rst [hunk 11]
```diff
@@ -708,14 +836,49 @@ You may want to use more than one mailer transport for delivery of your messages
...
+        # config/packages/mailer.yaml
+        framework:
+            mailer:
+                transports:
+                    main: '%env(MAILER_DSN)%'
+                    alternative: '%env(MAILER_DSN_IMPORTANT)%'
+
+    .. code-block:: xml
+
+        <!-- config/packages/framework.xml -->
```
Comment (xabbuh) [inline mailer.rst:852]: `<!-- config/packages/mailer.xml -->`

File: mailer.rst [hunk 12]
```diff
@@ -708,14 +836,49 @@ You may want to use more than one mailer transport for delivery of your messages
...
+                <framework:mailer>
+                    <framework:transport name="main">%env(MAILER_DSN)%</framework:transport>
+                    <framework:transport name="alternative">%env(MAILER_DSN_IMPORTANT)%</framework:transport>
+                </framework:mailer>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
```
Comment (xabbuh) [inline mailer.rst:872]: `// config/packages/mailer.php`

**Comment by wouterj** [comment]
Oh, I'm sorry for the countless copy-paste mistakes. Thanks for your detailed review!

**Review by xabbuh — APPROVED** [review]

**Comment by fabpot** [comment]
The change upstream has been merged now.

**Comment by xabbuh** [comment]
Thank you Wouter.

**Comment by oxan** [comment]
This change completely removed [this paragraph](https://github.com/symfony/symfony-docs/pull/13945/files#diff-15f60fe9b0fb24da3a2455c72e02c02aL88) from the documentation.

**Comment by wouterj** [comment]
Hi @oxan! The null DSN is documented here: https://symfony.com/doc/current/mailer.html#disabling-delivery

**Comment by oxan** [comment]
@wouterj Sorry, I was too curt with my comment. What I meant was that the sendmail DSN isn't documented anymore.

**Comment by wouterj** [comment]
No problem @oxan, thanks for clarifying :) I think I removed it in favor of this section https://symfony.com/doc/current/mailer.html#transport-setup . That section doesn't explicitly say "sendmail", but what it shows (native SMTP) is in fact sendmail. If you think we should be more explicit about it, please consider hitting the "edit this page" button at the top right. (btw, from my understanding, `smtp://` and `smtp+sendmail://` are equal)

**Comment by oxan** [comment]
@wouterj As I understand the code (I haven't actually verified this), `smtp://` will connect to an SMTP server on localhost, while `smtp+sendmail://` will invoke `sendmail` and talk SMTP over stdin to it (so it isn't necessary to have an SMTP server running). It's also possible to invoke `sendmail` without talking SMTP to it, but counterintuitively that is not achieved by using `sendmail://`, but you need to explicitly specify the command as an option using `sendmail://?command=/usr/sbin/sendmail -oi -t` (which is also not documented). Furthermore, the note in the documentation that user, pass and port are optional implies that host isn't and you are required to specify a host.

I think it'd be ideal to have a subsection for every transport along with the available options, but that'd be a quite involved change.

**Pull request merged**